pennylane/drawer/utils.py

```diff
@@ -103,70 +103,57 @@ def unwrap_controls(op):
+    >>> cwire_wires
+    [[0, 0, 3], [1, 0]]
+
+    From this information, we can see that the first classical wire is active in layers
+    0, 2, and 3 while the second classical wire is active in layers 1 and 2.  The first "active"
+    layer will always be the one with the mid circuit measurement.
 
-    # Map between mid-circuit measurements and their position on the drawing
-    # The bit map only contains mid-circuit measurements that are used in
-    # classical conditions.
+    """
     bit_map = {}
-
-    # Map between classical bit positions and the layer of their corresponding mid-circuit
-    # measurements.
-    measurement_layers = []
-
-    # Map between classical bit positions and the final layer where the bit is used.
-    # This is needed to know when to stop drawing a bit line. The bit is the index,
-    # so each of the two lists must have the same length as the number of bits
-    final_cond_layers = []
-
-    measurements_for_conds = set()
-    conditional_ops = []
-    for op in operations:
-        if isinstance(op, Conditional):
-            measurements_for_conds.update(op.meas_val.measurements)
-            conditional_ops.append(op)
-
-    if len(measurements_for_conds) > 0:
-        cond_mid_measures = [op for op in operations if op in measurements_for_conds]
-        cond_mid_measures.sort(key=operations.index)
-
-        bit_map = dict(zip(cond_mid_measures, range(len(cond_mid_measures))))
-
-        n_bits = len(bit_map)
-
-        # Set lists to correct size
-        measurement_layers = [None] * n_bits
-        final_cond_layers = [None] * n_bits
-
-        for i, layer in enumerate(layers):
-            for op in layer:
-                if isinstance(op, MidMeasureMP) and op in bit_map:
-                    measurement_layers[bit_map[op]] = i
-
-                if isinstance(op, Conditional):
-                    for mid_measure in op.meas_val.measurements:
-                        final_cond_layers[bit_map[mid_measure]] = i
-
-    return bit_map, measurement_layers, final_cond_layers
+    for layer in layers:
+        for op in layer:
+            if isinstance(op, Conditional):
+                for m in op.meas_val.measurements:
+                    bit_map[m] = None  # place holder till next pass
+
+    connected_layers = [[] for _ in bit_map]
+    connected_wires = [[] for _ in bit_map]
+    num_cwires = 0
+    for layer_idx, layer in enumerate(layers):
+        for op in layer:
+            if isinstance(op, MidMeasureMP) and op in bit_map:
+                bit_map[op] = num_cwires
+                connected_layers[num_cwires].append(layer_idx)
+                connected_wires[num_cwires].append(op.wires[0])
+                num_cwires += 1
+            elif isinstance(op, Conditional):
+                for m in op.meas_val.measurements:
+                    cwire = bit_map[m]
+                    connected_layers[cwire].append(layer_idx)
+                    connected_wires[cwire].append(max(op.wires))
```

Comment:
Just a note: this max value is not valid with custom wire labels. Shouldn't be a problem with `tape_mpl` as you're using `map_wires` before this function gets used, and it isn't a problem in `tape_text` because we don't use `connected_wires`.

I'd suggest leaving a comment here to note for future development. Maybe in the future it might need to be fixed (although it probably makes sense to map wires in `tape_text` first before considering fixing this). Maybe a keyword argument `wire_map` which we use if provided or assume that custom labels haven't been used if `wire_map` is `None`? Not to worry about it right now though.

Reply:
Might be something to re-examine in the future, but I think we should be fine for now.